reinforcement_learning/rlclientlib/vw_model/vw_model.cc

```diff
@@ -35,26 +35,28 @@ namespace reinforcement_learning { namespace model_management {
       pooled_vw vw(_vw_pool, _vw_pool.get_or_create());
 
       // Rank actions using the model.  Should generate a pdf
-      std::vector<int> actions;
-      std::vector<float> scores;
-      vw->rank(features, actions, scores);
+      std::vector<int> action_ids;
+      std::vector<float> pdf;
 
-      // Pick using the pdf. NOTE: sample_after_normalizing() can change the pdf
-      uint32_t action;
-      auto const scode = e::sample_after_normalizing(rnd_seed, std::begin(scores), std::end(scores), action);
+      // Get a ranked list of action_ids and corresponding pdf
+      vw->rank(features, action_ids, pdf);
+
+      // Pick a slot using the pdf. NOTE: sample_after_normalizing() can change the pdf
+      uint32_t chosen_index;
+      auto const scode = e::sample_after_normalizing(rnd_seed, std::begin(pdf), std::end(pdf), chosen_index);
 
       if ( S_EXPLORATION_OK != scode ) {
         RETURN_ERROR_LS(_trace_logger, status, exploration_error) << scode;
       }
 
-      response.push_back(actions[action], scores[action]);
+      response.push_back(action_ids[chosen_index], pdf[chosen_index]);
 
       // Setup response with pdf from prediction and chosen action
-      for (size_t idx = 1; idx < actions.size(); ++idx) {
-        const auto cur_idx = action != idx ? idx : 0;
-        response.push_back(actions[cur_idx], scores[cur_idx]);
+      for (size_t idx = 1; idx < action_ids.size(); ++idx) {
+        const auto cur_idx = chosen_index != idx ? idx : 0;
+        response.push_back(action_ids[cur_idx], pdf[cur_idx]);
       }
-      response.set_chosen_action_id(action);
+      response.set_chosen_action_id(action_ids[chosen_index]);
```

Comment:
This is the actual change.  The rest of the changes are renames and comments to clarify the code.